src/node_buffer.cc

```diff
@@ -1442,6 +1442,83 @@ void CopyArrayBuffer(const FunctionCallbackInfo<Value>& args) {
   memcpy(dest, src, bytes_to_copy);
 }
 
+// TODO(ronag): simdutf for this.
+// Write as many valid ascii bytes as possible to dst and return
+// the number of bytes written.
+uint32_t write_ascii(uint8_t* dst, const uint8_t* src, uint32_t size) {
+  size_t i = 0;
+
+  // 256 bit
+  for (; i + 32 <= size; i += 32) {
+    uint64_t f = 0;
+    uint64_t v = 0;
+    memcpy(&v, src + i, 8);
+    f |= v;
+    memcpy(&v, src + i + 8, 8);
+    f |= v;
+    memcpy(&v, src + i + 16, 8);
+    f |= v;
+    memcpy(&v, src + i + 24, 8);
+    f |= v;
+    if ((f & UINT64_C(0x8080808080808080)) != 0) {
+      break;
+    }
+    memcpy(dst + i, src + i, 32);
+  }
+
+  // 64 bit
+  for (; i + 8 <= size; i += 8) {
+    uint64_t v = 0;
+    memcpy(&v, src + i, 8);
+    if ((v & UINT64_C(0x8080808080808080)) != 0) {
+      break;
+    }
+    memcpy(dst + i, src + i, 8);
+  }
+
+  // 8 bit
+  for (; i + 1 <= size; i += 1) {
+    if ((src[i] & 0x80) != 0) {
+      break;
+    }
+    dst[i] = src[i];
+  }
+
+  return i;
+}
+
+uint32_t WriteUTF8String(v8::Isolate* isolate,
+                         uint8_t* dst,
+                         uint32_t dst_len,
+                         const uint8_t* src,
+                         uint32_t src_len) {
+  const auto size = std::min(src_len, dst_len);
+
+  // When writing ascii then written == read.
+  auto written = write_ascii(dst, src, size);
+
+  if (written < size) {
+    HandleScope scope(isolate);
+
+    // TODO(ronag): Unfortunately this performs unnecessary allocation
+    // and copy. Is there a way to StringBytes::Write without a String?
+    const auto str =
+        String::NewFromOneByte(isolate,
+                               src + written,
+                               NewStringType::kNormal,
+                               src_len - written)
```

Comment:
Is there some kind of `String:ViewFromOneByte` method to avoid the backing store allocation and copy?

Reply:
Alternatively a `StringBytes::Write` overload that takes a `{ char* ptr; size_t length; }`  instead of `String`.

Reply:
I mean, like the comment above says, you're really just looking for `convert_latin1_to_utf8` with a string delimiter. `StringBytes` is really just intended for when you work wigth a JS `String` to begin with

Reply:
@lemire is that something we can add?

Reply:
@addaleax how about now?